modules/electrophysiology_browser/php/models/electrophysioevents.class.inc

```diff
@@ -210,51 +286,95 @@ class ElectrophysioEvents
             ['PFID' => $this->_physioFileID]
         );
 
-        if (count($tsv) > 0) {
-            $tsvPath = $dataDir . $tsv[0]['filePath'];
-            // Update files if files updated before database updated
-            if ($tsv[0]['lastWritten'] <= $tsv[0]['lastUpdate']) {
-                // events.tsv
-                $tsvFile = fopen($tsvPath, 'w'); // Will override all file content
+        $projectID = intval(
+            $db->pselectOne(
+                'SELECT ProjectID FROM session AS s WHERE s.ID = (
+                SELECT SessionID FROM physiological_file
+                WHERE PhysiologicalFileID=:PFID
+            )',
+                ['PFID' => $this->_physioFileID]
+            )
+        );
 
-                $extraColumns = $db->pselect(
-                    "SELECT *
+        $json = $db->pselect(
+            "SELECT
+        EventFileID   AS id,
+        FilePath      AS filePath,
+        LastUpdate    AS lastUpdate,
+        LastWritten   AS lastWritten
+      FROM physiological_event_file
+      WHERE (
+          PhysiologicalFileID=:PFID OR ProjectID=:PID
+      )
+      AND FileType='json'",
+            [
+                'PFID' => $this->_physioFileID,
+                'PID'  => $projectID
+            ]
+        );
+
+        $tsvPath  = count($tsv) > 0 ? $dataDir . $tsv[0]['filePath'] : '';
+        $jsonPath = count($json) > 0 ? $dataDir . $json[0]['filePath'] : '';
+
+        // Update files if files updated before database updated
+        if ($tsv[0]['lastWritten'] <= $tsv[0]['lastUpdate']
+            || $json[0]['lastWritten'] <= $json[0]['lastUpdate']
+        ) {
+            // events.tsv
+            $tsvFile = fopen($tsvPath, 'w'); // Will override all file content
```

Comment:
I need to review this part much more closely but I'm submitting the rest of the review for now.